mriqc/utils/bids.py

```diff
@@ -32,3 +35,62 @@ def collect_bids_data(
         )
 
     return imaging_data
+
+
+def write_bidsignore(deriv_dir):
+    bids_ignore = (
+        "*.html", "logs/",  # Reports
+        "*_T1w.json", "*_T2w.json", "*_bold.json",  # Outputs are not yet standardized
```

Comment:
aren't these standard?

Reply:
There's no standardization of QC measures at this point. These would be expected to apply to `.nii.gz` files, but there are none to apply to, so I suspect (but haven't checked) that the validator would get upset. Under normal circumstances, `bold.json` at least requires `TaskName` and `RepetitionTime` (or other timing information), which aren't present.

Reply:
> These would be expected to apply to `.nii.gz` files, but there are none to apply to

Perhaps BIDS-Derivatives are missing a mechanism to add metadata to a raw file? This could be hard to implement and presents the problem of how you decide the hierarchy when two derivative folders attempt to set the same metadata or if you attempt to shadow some existing metadata in the raw dataset. But for MRIQC and descriptive tools as such, it would be killer.

Reply:
This feels easy to declare, not sure about implement. @tyarkoni probably has the clearest perspective on how easy/painful that's likely to be.

Reply:
Not sure what the scope of the suggestion is... Just having a standalone utility that takes an arbitrary filename or `BIDSFile` and writes an entry to the JSON file seems trivial. Is that the idea?

If you want to do it via the `BIDSLayout`, so that the indexing automatically syncs with the new metadata (instead of having to re-index the whole project), that would still be pretty straightforward, I think, though I'd have to think about it some more to be sure. In essence you could just have a `BIDSFile.update_metadata()` class method that sets new metadata once the file-writing operation has successfully completed. The SQLAlchemy query-building might be a bit of a pain in the ass, because you'd need to check if there are any new entities, and if so, create DB records for them, but I don't think there would be any serious difficulties.

Is either of those what you have in mind, or are you thinking of something else?

Reply:
> This could be hard to implement and presents the problem of how you decide the hierarchy when two derivative folders attempt to set the same metadata or if you attempt to shadow some existing metadata in the raw dataset

I don't think there's any ambiguity here, at least as I read/think about the spec. If you have two separate derivatives folders, they can have conflicting metadata; I don't see anything that implies any global harmonization. And if you have a chain of derivatives (or a derivative shadowing raw), then I think it's natural to expect later changes to always overwrite earlier ones (just like inheritance).

Reply:
@tyarkoni I'm having trouble parsing your post right now, so this is to clarify what I understood as Oscar's proposal:

```
bids/
    derivatives/mriqc/sub-01/anat/
        sub-01_T1w.json
    sub-01/anat/
        sub-01_T1w.nii.gz  <- MRIQC metadata applies to this data
```

Possibly this was entirely clear to you and I'm too tired to understand your points. Will re-read tomorrow.

Reply:
@effigies that's what I understood as well; I'm just trying to figure out what the proposal would entail from a PyBIDS perspective, on the assumption that was the reason for tagging me. If this is just a question about the spec, I don't think I have any useful insight; at first blush, I don't think the spec needs to say anything more than it currently does in order to allow users to add arbitrary metadata to any file.